src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/BreakpointEvidence.java

```diff
@@ -51,6 +51,12 @@ protected BreakpointEvidence( final Kryo kryo, final Input input ) {
     public boolean isValidated() { return validated; }
     public void setValidated( final boolean validated ) { this.validated = validated; }
 
+    // default interfaces to be overridden by derived classes where these values are defined
+    public Integer getMappingQuality() { return null; }
```

Comment:
Would it work to use `OptionalInt` and `Optional<String>`?  The advantages are that 1) `.isPresent()` is much clearer than ` == null` and 2) the fact that the type is `Optional` makes it self-documenting that it's not necessarily there.

Reply:
@TedBrookings Even though we talked about this briefly today, I'm looking at these again tonight and wondering why we don't just put them on the appropriate subtypes? Ie. put mapping quality, template size, and cigar string on `ReadEvidence` and read count on `TemplateSizeAnomaly`? Then rather than check for null in `XGBoostEvidenceFilter` check the class type using `instanceof` and get the appropriate values?

Reply:
That's how I originally wrote it. It was requested that I use polymorphism and Optional values instead (I used null because that's what was used in prior BreakpointEvidence code). I'm happy to do whatever. Between notes, and conversations, and additional mulling; these are the options I see on the table:
1) use instanceof
2) polymorphism, return null if undefined, check != null
3) polymorphism, return Optional, check Optional::ispresent
4) polymorphism, write checker function e.g. templateSizeIsDefined() but only define getTemplateSize() in subclasses where templateSizeIsDefined() is true
5) polymorphism, write BreakpointEvidence::getFeatures() member function and override. Since overlap and other multi-evidence features are needed, BreakpointEvidence::getFeatures() will be a starting point for computation rather than the end, but it could handle all the NaN logic and return a BreakpointEvidenceFeatures object that's reasonably self-documenting.

I'm totally happy to do any of 1-5 but I think I'd like an official call on this one so I'm not going-round in circles.

Reply:
Let me make a suggestion:
I think that the "smelly" aspect of the current code -- the part that seems not quite properly object oriented -- is testing for a particular concrete class:  ReadEvidence.
This should be an interface, rather than a concrete class.  Asking a piece of evidence, "Are you the kind of evidence that's associated with a particular read?" is OK in my book.  (And then you downcast to the interface that allows you to interrogate properties of the associated read.)
So I'd recommend splitting the current ReadEvidence class into an interface and a concrete implementation.  I should have done this to begin with.  Sorry I took a shortcut.

Reply:
@tedsharpe's suggestion sounds good to me.  I'm dropping out of this one so as to avoid too many cooks spoiling broth.

Reply:
This is a pretty minor debate over a few lines of code -- OO purists would tell us to implement a Visitor pattern but that's way too much work and would too tightly couple the features needed for the classifier to the classifier IMO. @tedsharpe 's proposal sounds like a good compromise. I can also see the use of `Optionals` in this case. @TedBrookings  I'm usually fine with checking `nulls`, as I said yesterday, but in this case where null is being used for a subclass-specific data field (and therefore sort of overriding the inheritance hierarchy of the class) it does seem pretty ugly.